### DISCUSSION TITLE
3.0.3.9

### DISCUSSION THREAD

**Discussion by danielkerr** [opencart/opencart#12784] [Announcements]
## What's Changed
* fixed install text issue by @WebkulOpencart in https://github.com/opencart/opencart/pull/10179
* Added HTTPS on docs.opencart.com HREF - admin/view/template/common/header.twig file by @TheCartpenter in https://github.com/opencart/opencart/pull/10178
* Fix for bug #10197 in maintenance branch v2 by @mathsolutions in https://github.com/opencart/opencart/pull/10203
* Fix for issue #10216 by @mathsolutions in https://github.com/opencart/opencart/pull/10217
* Corrected typo - admin/controller/common/filemanager.php file by @TheCartpenter in https://github.com/opencart/opencart/pull/10171
* Fixed deleteApiSessionBySessionId method call - admin/controller/extension/payment/klarna_checkout.php file by @TheCartpenter in https://github.com/opencart/opencart/pull/10135
* Pushed dots on $code - admin/controller/design/translation.php file by @TheCartpenter in https://github.com/opencart/opencart/pull/10127
* Added space on IF statements - admin/controller/common/login.php file by @TheCartpenter in https://github.com/opencart/opencart/pull/10116
* Added space on IF statements - admin/controller/marketing/contact.php file by @TheCartpenter in https://github.com/opencart/opencart/pull/10117
* Added space on IF statement - admin/model/localisation/currency.php file by @TheCartpenter in https://github.com/opencart/opencart/pull/10118
* Added space on IF statement - catalog/controller/event/activity.php file by @TheCartpenter in https://github.com/opencart/opencart/pull/10119
* fixed the issue https://github.com/opencart/opencart/issues/9739 by @webkul-dilip in https://github.com/opencart/opencart/pull/10248
* Improved summernote  language  by @webkul-dilip in https://github.com/opencart/opencart/pull/10275
* [3.0.x.x] Fixed custom_fields not inserted in database when register as affiliate by @condor2 in https://github.com/opencart/opencart/pull/10322
* [3.0.x.x] Autogenerate tracking code when add a customer from admin by @condor2 in https://github.com/opencart/opencart/pull/10323
* Fix upload date filter by @ADDCreative in https://github.com/opencart/opencart/pull/10326
* [3.0.x.x] Update affiliate custom_field when edit a customer by @condor2 in https://github.com/opencart/opencart/pull/10325
* Split base64 encoded lines in mail by @ADDCreative in https://github.com/opencart/opencart/pull/10269
* Fix mail line endings for PHP 8 by @ADDCreative in https://github.com/opencart/opencart/pull/10539
* Don't use strftime as deprecated in PHP 8.1 by @ADDCreative in https://github.com/opencart/opencart/pull/10457
* Change session read return type to array by @ADDCreative in https://github.com/opencart/opencart/pull/10414
* Fix deprecated optional parameter in PHP 8 by @ADDCreative in https://github.com/opencart/opencart/pull/10541
* Fix null credentials in marketplace for PHP 8 by @ADDCreative in https://github.com/opencart/opencart/pull/10542
* PHP 8.x support for 3.0.x.x maintenance branch (09-03-2022) by @mhcwebdesign in https://github.com/opencart/opencart/pull/10467
* [Maintenance] - Removed console.log entry - catalog/view/theme/default/template/extension/advertise/google_dynamic_remarketing_cart.twig file by @TheCartpenter in https://github.com/opencart/opencart/pull/10674
* [3.0.x.x] Fixed PHP 8.1 warning pp_braintree by @condor2 in https://github.com/opencart/opencart/pull/10679
* [3.0.x.x]Fixed PHP 8.1 warning marketplace extension info by @condor2 in https://github.com/opencart/opencart/pull/10682
* 3.0.x.x maintenance - update scssphp for PHP8.x by @mhcwebdesign in https://github.com/opencart/opencart/pull/10684
* [Maintenance] - Removed console.log entry - catalog/view/theme/default/template/extension/payment/paypal.twig file by @TheCartpenter in https://github.com/opencart/opencart/pull/10677
* [Maintenance] - Removed console.log entry - catalog/view/theme/default/template/extension/advertise/google_dynamic_remarketing_product.twig file by @TheCartpenter in https://github.com/opencart/opencart/pull/10675
* [Maintenance] - Replaced addReference with editReference - catalog/controller/extension/payment/pp_express.php file by @TheCartpenter in https://github.com/opencart/opencart/pull/10681
* Fix error handler for PHP 8 by @ADDCreative in https://github.com/opencart/opencart/pull/10835
* Fix passed by reference warning in PHP 8 by @ADDCreative in https://github.com/opencart/opencart/pull/10836
* [Maintenance] - Added payment prefixes - admin/view/template/extension/payment/klarna_account.twig file by @TheCartpenter in https://github.com/opencart/opencart/pull/10815
* Revert change made to contact from address by @ADDCreative in https://github.com/opencart/opencart/pull/8931
* [3.0.x.x] Fixed sort by username API by @condor2 in https://github.com/opencart/opencart/pull/11138
* [3.0.x.x] Added missing language variable Menu by @condor2 in https://github.com/opencart/opencart/pull/11153
* PHP 8.x support for catalog/model/product.php by @mhcwebdesign in https://github.com/opencart/opencart/pull/11430
* Fixed custom fields support for affiliate by @mhcwebdesign in https://github.com/opencart/opencart/pull/11463
* Fix validation for custom_fields in checkout by @MarvinKlein1508 in https://github.com/opencart/opencart/pull/11464
* Refresh left menu and extensions list after an extension uninstall by @mhcwebdesign in https://github.com/opencart/opencart/pull/11669
* Load correct language file in install function. by @mhcwebdesign in https://github.com/opencart/opencart/pull/11670
* 3.0.x.x maintenance by @mhcwebdesign in https://github.com/opencart/opencart/pull/11680
* Fixed pagination. by @mhcwebdesign in https://github.com/opencart/opencart/pull/11817
* 3.0.x.x maintenance by @mhcwebdesign in https://github.com/opencart/opencart/pull/11874
* [Maintainance] Fix onRegionClick by @osworx in https://github.com/opencart/opencart/pull/11880
* mail classes fixes for PHP 8.2 by @mhcwebdesign in https://github.com/opencart/opencart/pull/12070
* Bugfix for using str_split in PHP 8.2 by @mhcwebdesign in https://github.com/opencart/opencart/pull/12073
* [3.0.x.x] Fixed PHP 8.2 recurring warning by @condor2 in https://github.com/opencart/opencart/pull/12087
* [3.0.x.x_Maintenance] Add support for images of type *.webp by @nmpetkov in https://github.com/opencart/opencart/pull/12104
* [3.0.x.x] Added webp support filemanager controller by @condor2 in https://github.com/opencart/opencart/pull/12109
* [3.0.x.x] Added webp support opencart.sql by @condor2 in https://github.com/opencart/opencart/pull/12110
* Bugfix order editor for special characters in form fields by @mhcwebdesign in https://github.com/opencart/opencart/pull/12116
* Keep order options sort order by @mhcwebdesign in https://github.com/opencart/opencart/pull/12164
* Fix double slash in session path by @ADDCreative in https://github.com/opencart/opencart/pull/10536
* Italic support for summernote by @mhcwebdesign in https://github.com/opencart/opencart/pull/12210
* 3.0.x.x maintenance by @mhcwebdesign in https://github.com/opencart/opencart/pull/12230
* bugfix backup / restore by @mhcwebdesign in https://github.com/opencart/opencart/pull/12296
* Check if encryption negotiation failed by @ADDCreative in https://github.com/opencart/opencart/pull/12321
* [3.0.x.x] Update pagination Return controller by @condor2 in https://github.com/opencart/opencart/pull/12252
* [3.0.x.x] Update Download pagination controller by @condor2 in https://github.com/opencart/opencart/pull/12253
* [3.0.x.x] Update Order pagination controller by @condor2 in https://github.com/opencart/opencart/pull/12254
* [3.0.x.x] Update Pagination in Transaction controller by @condor2 in https://github.com/opencart/opencart/pull/12255
* [3.0.x.x] Update pagination code Reward controller by @condor2 in https://github.com/opencart/opencart/pull/12256
* [3.0.x.x]Update paginatio Recurring controler by @condor2 in https://github.com/opencart/opencart/pull/12257
* Update city length check by @dennisbuijsgmailcom in https://github.com/opencart/opencart/pull/12251
* no more addthis by @mhcwebdesign in https://github.com/opencart/opencart/pull/12362
* matching default fontsizes by @mhcwebdesign in https://github.com/opencart/opencart/pull/12385
* Fix SMTP line lengths by @ADDCreative in https://github.com/opencart/opencart/pull/12384
* 3.0.x.x maintenance by @mhcwebdesign in https://github.com/opencart/opencart/pull/12388
* 3.0.x.x maintenance by @mhcwebdesign in https://github.com/opencart/opencart/pull/12390
* 3.0.x.x maintenance by @mhcwebdesign in https://github.com/opencart/opencart/pull/12391
* Update Twig docs link by @bloodyKnuckles in https://github.com/opencart/opencart/pull/12156
* need leading space to prevent appending to link by @daretogo in https://github.com/opencart/opencart/pull/12223
* 3.0.x.x maintenance by @mhcwebdesign in https://github.com/opencart/opencart/pull/12463
* filemanager fix by @mhcwebdesign in https://github.com/opencart/opencart/pull/12491
* deprecation error fixes by @mhcwebdesign in https://github.com/opencart/opencart/pull/12519
* Update translation_form.twig by @mhcwebdesign in https://github.com/opencart/opencart/pull/12640
* Fixed error displaying guest shipping by @SirStepheno in https://github.com/opencart/opencart/pull/12672
* re-added old jquery by @mhcwebdesign in https://github.com/opencart/opencart/pull/12715

## New Contributors
* @mathsolutions made their first contribution in https://github.com/opencart/opencart/pull/10203
* @bloodyKnuckles made their first contribution in https://github.com/opencart/opencart/pull/12156
* @daretogo made their first contribution in https://github.com/opencart/opencart/pull/12223
* @SirStepheno made their first contribution in https://github.com/opencart/opencart/pull/12672

**Full Changelog**: https://github.com/opencart/opencart/compare/3.0.3.8...3.0.3.9

<hr /><em>This discussion was created from the release <a href='https://github.com/opencart/opencart/releases/tag/3.0.3.9'>3.0.3.9</a>.</em>

**Comment by DanielHatano** [comment]
Opencart 3 Still Alive!!!!!!  
Good luck Opencart 4 without OCMOD  🤣🤣

**Reply by felipoantonoff** [reply]
Opencart 4 should be back with OCMOD soon :)

**Comment by Gadalla** [comment]
Is there any particular reason to go back to version 3? And does it have OCMOD?

**Reply by IP-CAM** [reply]
The likely main reason was it's popularity, combined with the amount of Extensions.

**Reply by felipoantonoff** [reply]
I think it's because Opencart 4 is still maturing and because Opencart 3 is more used and has more Modules.

I hope that in 2024 Opencart 4 will be the main and most used version of Opencart and of course it will return with OCMOD forever.

**Reply by Gadalla** [reply]
"I hope that in 2024 Opencart 4 will be the main and most used version of Opencart and of course it will return with OCMOD forever."
I hope so. And they introduce new features like countries and zones in multiple languages, bundle products, discounts on QTY, discounts for category, etc.

**Comment by danielkerr** [comment]
yes will add it on the project to do list.

**Comment by mhcwebdesign** [comment]
@danielkerr: What's the reason for having both a 3.0.x.x_Maintenance and 3.0.x.x branch? Shouldn't  the 3.0.x.x be sufficient for future bugfixes ? 3.0.x.x already is based on 3.0.3.9, with some further small bugfixes.

**Comment by danielkerr** [comment]
i renamed it

**Comment by mhcwebdesign** [comment]
OK, thanks.
